src/agent/solana/exporter.rs

```diff
@@ -722,12 +722,20 @@ impl Exporter {
 
             let result = result_rx.await??;
 
-            // Calculate the maximum slot difference between aggregate slot and
+            // Calculate the maximum slot difference between the publisher latest slot and
             // current slot amongst all the accounts. Here, the aggregate slot is
             // used instead of the publishers latest update to avoid overpaying.
             let oldest_slot = result
                 .values()
-                .map(|account| account.last_slot)
+                .flat_map(|account| {
+                    account
+                        .comp
+                        .iter()
+                        .filter(|c| c.publisher == publish_keypair.pubkey())
+                        .collect::<Vec<_>>()
+                        .get(0)
+                        .map(|c| c.latest.pub_slot)
+                })
```

Comment:
Nitpick you can ignore as I know you want to move this faster, but think this is equivelent to just:

```
account
    .comp
    .iter()
    .find(|c| c.publisher == publish_keypair.pubkey())
    .map(|c| c.latest.pub_slot)
```

Are you sure this won't fail with a panic!() in your current code on the `.get(0)`? I trust the assumption is they wouldn't be looking at the feed if not, but just want to confirm?